test/ao/sparsity/test_structured_sparsifier.py

```diff
@@ -916,3 +917,421 @@ def test_prune_lstm_layernorm_linear_single_layer(self):
         # also check that output of linear is the same shape, this means we've resized
         # linear columns correctly.
         assert out_expected.shape == out_pruned.shape
+
+class TestFPGMSparsifier(TestCase):
```

Comment:
It's not necessary to have all these specific pattern checks, as they are already being checked in test_sparse_semi_structured (for the base class). I would remove them, as it'll just add extra time for CI and not provide much signal. 

At the same time I don't see an explicit test for the implementation of FPGM? I'd like to see a test that makes sure `update_mask` is implemented correctly, like [here](https://github.com/pytorch/pytorch/blob/33b855e9069ddc98399c0dfd362f18df9503b66f/test/ao/sparsity/test_structured_sparsifier.py#L75). This should be the only test you need.

Reply:
A test for _compute_distances would also be nice